cps/kni.c

```diff
@@ -381,72 +381,121 @@ attr_get(struct cps_config *cps_conf,
 			RTE_LOG(INFO, GATEKEEPER, "cps update: multipath\n");
 	}
 	if (tb[RTA_DST]) {
-		if (family == AF_INET6) {
+		if (family == AF_INET) {
+			struct in_addr *addr =
+				mnl_attr_get_payload(tb[RTA_DST]);
+			rte_memcpy(&update->ip.v4,
+				addr, sizeof(update->ip.v4));
+			if (cps_conf->debug)
+				inet_print("dst", addr);
+		} else if (likely(family == AF_INET6)) {
 			struct in6_addr *addr =
 				mnl_attr_get_payload(tb[RTA_DST]);
+			rte_memcpy(&update->ip.v6,
+				addr, sizeof(update->ip.v6));
 			if (cps_conf->debug)
 				inet6_print("dst", addr);
 		} else {
-			struct in_addr *addr =
-				mnl_attr_get_payload(tb[RTA_DST]);
-			if (cps_conf->debug)
-				inet_print("dst", addr);
+			RTE_LOG(WARNING, GATEKEEPER,
+				"cps update: unknown address family %d at %s!\n",
+				family, __func__);
+			return;
 		}
+
+		dst_present = true;
 	}
 	if (tb[RTA_SRC]) {
-		if (family == AF_INET6) {
+		RTE_LOG(WARNING, GATEKEEPER,
+			"cps update: the rtnetlink command has information (RTA_SRC) that we don't need or don't honor!\n");
+
+		if (family == AF_INET) {
+			struct in_addr *addr =
+				mnl_attr_get_payload(tb[RTA_SRC]);
+			if (cps_conf->debug)
+				inet_print("src", addr);
+		} else if (likely(family == AF_INET6)) {
 			struct in6_addr *addr =
 				mnl_attr_get_payload(tb[RTA_SRC]);
 			if (cps_conf->debug)
 				inet6_print("src", addr);
 		} else {
-			struct in_addr *addr =
-				mnl_attr_get_payload(tb[RTA_SRC]);
-			if (cps_conf->debug)
-				inet_print("src", addr);
+			RTE_LOG(WARNING, GATEKEEPER,
+				"cps update: unknown address family %d at %s!\n",
+				family, __func__);
+			return;
 		}
 	}
 	if (tb[RTA_OIF]) {
+		update->oif_index = mnl_attr_get_u32(tb[RTA_OIF]);
+
 		if (cps_conf->debug)
 			RTE_LOG(INFO, GATEKEEPER, "cps update: oif=%u\n",
-				mnl_attr_get_u32(tb[RTA_OIF]));
+				update->oif_index);
 	}
 	if (tb[RTA_FLOW]) {
+		RTE_LOG(WARNING, GATEKEEPER,
+			"cps update: the rtnetlink command has information (RTA_FLOW) that we don't need or don't honor!\n");
+
```

Comment:
Incorporate `mnl_attr_get_u32(tb[RTA_FLOW])` into the warning message and rop the `if` statement below.

Analogous change applies to `RTA_PRIORITY`.